src/lib/utils/options/declaration.ts

```diff
@@ -276,66 +313,86 @@ export function convert<T extends DeclarationOption>(
-                    numberOption.minValue,
-                    numberOption.maxValue
-                )
+                !valueIsWithinBounds(numValue, option.minValue, option.maxValue)
             ) {
                 throw new Error(
                     getBoundsError(
-                        numberOption.name,
-                        numberOption.minValue,
-                        numberOption.maxValue
+                        option.name,
+                        option.minValue,
+                        option.maxValue
                     )
                 );
             }
+            if (option.validate) {
+                option.validate(numValue);
+            }
             return numValue;
         }
-        case ParameterType.Boolean:
-            return Boolean(value);
-        case ParameterType.Array:
+        case ParameterType.Boolean: {
+            const boolValue = Boolean(value);
+            if (option.validate) {
+                option.validate(boolValue);
+            }
+            return boolValue;
+        }
+        case ParameterType.Array: {
+            let strArrValue = new Array<string>();
             if (Array.isArray(value)) {
-                return value.map(String);
+                strArrValue = value.map(String);
             } else if (typeof value === "string") {
-                return value.split(",");
+                strArrValue = value.split(",");
+            }
+            if (option.validate) {
+                option.validate(strArrValue);
             }
-            return [];
+            return strArrValue;
+        }
         case ParameterType.Map: {
-            const optionMap = option as MapDeclarationOption<unknown>;
             const key = String(value).toLowerCase();
-            if (optionMap.map instanceof Map) {
-                if (optionMap.map.has(key)) {
-                    return optionMap.map.get(key);
+            let mapValue;
+            if (option.map instanceof Map) {
+                if (option.map.has(key)) {
+                    mapValue = option.map.get(key);
                 }
-                if ([...optionMap.map.values()].includes(value)) {
-                    return value;
+                if ([...option.map.values()].includes(value)) {
+                    mapValue = value;
                 }
             } else {
-                if (key in optionMap.map) {
-                    return optionMap.map[key];
+                if (key in option.map) {
+                    mapValue = option.map[key];
                 }
-                if (Object.values(optionMap.map).includes(value)) {
-                    return value;
+                if (Object.values(option.map).includes(value)) {
+                    mapValue = value;
                 }
             }
-            throw new Error(
-                optionMap.mapError ?? getMapError(optionMap.map, optionMap.name)
-            );
+            if (mapValue == undefined) {
```

Comment:
This isn't sufficient. Someone could conceivably have a map that looks like this: `{ null: null, undefined: undefined }` to change how missing values are output in JSON.

Reply:
Sorry, I don't get it.

Reply:
Yep, that looks good!